src/applications/vaos/services/appointment/index.js

```diff
@@ -119,47 +119,66 @@ function hasPartialResults(response) {
 export async function getBookedAppointments({
   startDate,
   endDate,
-  useV2 = false,
+  useV2VA = false,
+  useV2CC = false,
 }) {
   try {
-    if (useV2) {
-      const appointments = await getAppointments(startDate, endDate, [
-        'booked',
-        'cancelled',
-      ]);
-
-      const appointmentsWithoutRequests = appointments.filter(
-        appt => !appt.requestedPeriods,
+    let appointments = [];
+    if (!useV2VA) {
+      const confirmedVAAppointments = await getConfirmedAppointments(
+        'va',
+        moment(startDate).toISOString(),
+        moment(endDate).toISOString(),
       );
 
-      return transformVAOSAppointments(appointmentsWithoutRequests);
+      // We might get partial results back from MAS, so throw an error if we do
+      if (hasPartialResults(confirmedVAAppointments)) {
+        throw mapToFHIRErrors(
+          appointments[0].errors,
+          'MAS returned partial results',
+        );
+      }
+
+      appointments = [
+        ...transformConfirmedAppointments(confirmedVAAppointments.data),
+      ];
     }
 
-    const appointments = await Promise.all([
-      getConfirmedAppointments(
-        'va',
-        moment(startDate).toISOString(),
-        moment(endDate).toISOString(),
-      ),
-      getConfirmedAppointments(
+    if (!useV2CC) {
+      const confirmedCCAppointments = await getConfirmedAppointments(
         'cc',
         moment(startDate).toISOString(),
         moment(endDate).toISOString(),
-      ),
-    ]);
-
-    // We might get partial results back from MAS, so throw an error if we do
-    if (hasPartialResults(appointments[0])) {
-      throw mapToFHIRErrors(
-        appointments[0].errors,
-        'MAS returned partial results',
       );
+      appointments = [
+        ...appointments,
+        ...transformConfirmedAppointments(confirmedCCAppointments.data),
+      ];
+    }
+
+    if (useV2VA || useV2CC) {
+      const kindsOfVAAppointments = ['clinic', 'telehealth', 'phone'];
+      const allAppointments = await getAppointments(startDate, endDate, [
+        'booked',
+        'cancelled',
+      ]);
+
+      const filteredAppointments = allAppointments.filter(appt => {
+        if (
+          (!useV2VA && kindsOfVAAppointments.includes(appt.kind)) ||
```

Comment:
Would it work make this check `&& appt.kind !== 'cc'`?

Reply:
Good call, kind of overcomplicated this bit by creating the va kinds array.